jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -188,6 +188,17 @@ def cwd_for_path(self, path):
             os_path = os.path.dirname(os_path)
         return os_path
 
+    async def _remove_kernel_when_ready(self, kernel_id, kernel_awaitable):
+        try:
+            await super()._remove_kernel_when_ready(kernel_id, kernel_awaitable)
+        finally:
+            # Unlike its async sibling method in AsyncMappingKernelManager, removing the kernel_id
+            # from the connections dictionary isn't as problematic before the shutdown since the
+            # method is synchronous.  However, we'll keep the relative call orders the same from
+            # a maintenance perspective.
```

Comment:
This comment is (now) a bit confusing and I think it can be removed entirely, or replaced with something more applicable.

Reply:
I'm in favor of removing it.